src/format.rs

```diff
@@ -472,7 +472,29 @@ impl Printf for f32 {
 impl Printf for &str {
     fn format(&self, spec: &ConversionSpecifier) -> Result<String> {
         if spec.conversion_type == ConversionType::String {
-            Ok((*self).to_owned())
+            let mut s = String::new();
+
+            let width: usize = match spec.width {
+                NumericParam::Literal(w) => w,
+                _ => {
+                    return Err(PrintfError::Unknown); // should not happen at this point!!
+                }
+            }
+            .try_into()
+            .unwrap_or_default();
+
+            if spec.left_adj {
+                s.push_str(self);
+                while s.len() < width {
```

Comment:
It's a shame `String` has no `resize()` method like `Vec` does... (but I suppose that wouldn't work well with multi-byte characters)